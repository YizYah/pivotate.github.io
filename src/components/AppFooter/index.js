import React, { useState, useEffect } from 'react';
import { HashLink as Link } from 'react-router-hash-link';
// import { Context as AuthContext } from '../../context/AuthContext';

const AppFooter = (props) => {
  // const {state: {userLoggedIn}} = useContext(AuthContext)
  const [displaySignUp, setDisplaySignUp] = useState(false);

  useEffect(() => {
    setDisplaySignUp(localStorage.getItem('accessToken'));
  }, [displaySignUp]);

  return (
    <footer className='footer'>
      {!displaySignUp ? (
        <div className='footer__sign-up'>
          <div className='container triangle--white triangle'>
            <div className='flex'>
              <div className='grow-6'>
                <h2>
                  <div>Don't settle for</div>
                  <div>templates or freelancers!</div>
                </h2>
              </div>
              <div className='grow-6 text-center'>
                <button
                  type='button'
                  onClick={props.onSignUp}
                  href='#'
                  className='button button--large button--teal'
                >
                  SIGN UP <span className='chevron right' />
                </button>
              </div>
            </div>
          </div>
        </div>
      ) : null}
      <div className='footer__content'>
        <div className='container'>
          <div className='flex'>
            <div className=' grow-6'>
              <div>
                <img
                  height='70'
                  src='https://pivotatestaticassets.com/images/Pivotate Logo_White.svg'
                  alt='Pivotate Logo'
                />
              </div>
              <div>
                <Link to='/contact?s=footer'>Contact</Link>
              </div>
              <div className='powered-by'>
                <span className='grey--text'>Powered By</span>
                <strong href='#' className='teal--text'>
                  {' '}
                  NoStack
                </strong>
              </div>
            </div>
            <div className='grow-6 '>
              <div className='flex footer__group-links'>
                <div className='grow-4' />
                <div className='grow-4'>
                  <div className='footer__title'>About</div>
                  <ul className='footer__links'>
                    <li>
                      <Link to='/#about-us'> About Us</Link>
                    </li>
                    <li>
                      <Link to='/#how-it-works'> How it Works</Link>
                    </li>
                    <li>
                      <Link to='/pricing'> Pricing</Link>
                    </li>
                    <li>
                      {!displaySignUp ? (
                        <strong
                          role='button'
                          style={{ cursor: 'pointer' }}
                          tabIndex={0}
                          aria-hidden='true'
                          href='#'
                          onClick={props.onSignUp}
                          className='teal--text'
                        >
                          Sign Up Now
                        </strong>
                      ) : null}
                    </li>
                  </ul>
                </div>
                <div className='grow-4'>
                  <div className='footer__title'>Terms</div>
                  <ul className='footer__links'>
                    <li>
                      {' '}
                      <Link to='/terms-and-conditions'>Terms of Use</Link>
                    </li>
                    <li>
                      {' '}
                      <Link to='/privacy-policy'>Privacy</Link>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
          <div className='text-center copyright'>Copyright Â© 2020 Pivotate</div>
        </div>
      </div>
    </footer>
  );
};

export default AppFooter;
