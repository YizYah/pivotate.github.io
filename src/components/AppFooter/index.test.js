import { shallow } from 'enzyme';
import React from 'react';
import AppFooter from '.';

describe('Footer component', () => {
  beforeEach(() => {
    // jest
    //   .spyOn(global.localStorage, 'getItem')
    //   .mockImplementation((key) => store[key]);
    // jest
    //   .spyOn(global.localStorage, 'setItem')
    //   .mockImplementation((key, value) => {
    //     store[key] = value.toString();
    //   });
  });

  describe('AND there is no accessToken set', () => {
    beforeEach(() => {
      // store.
    });
    it('SHOULD render the footer with signup node', () => {
      const output = shallow(<AppFooter />);
      expect(output).toMatchSnapshot();
      expect(output.find('div.footer__sign-up').length).toBe(1);
    });
  });
});
