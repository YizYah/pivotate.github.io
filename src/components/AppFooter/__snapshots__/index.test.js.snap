// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component AND there is no accessToken set SHOULD render the footer with signup node 1`] = `
<footer
  className="footer"
>
  <div
    className="footer__sign-up"
  >
    <div
      className="container triangle--white triangle"
    >
      <div
        className="flex"
      >
        <div
          className="grow-6"
        >
          <h2>
            <div>
              Don't settle for
            </div>
            <div>
              templates or freelancers!
            </div>
          </h2>
        </div>
        <div
          className="grow-6 text-center"
        >
          <button
            className="button button--large button--teal"
            href="#"
            type="button"
          >
            SIGN UP 
            <span
              className="chevron right"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer__content"
  >
    <div
      className="container"
    >
      <div
        className="flex"
      >
        <div
          className=" grow-6"
        >
          <div>
            <img
              alt="Pivotate Logo"
              height="70"
              src="https://pivotatestaticassets.com/images/Pivotate Logo_White.svg"
            />
          </div>
          <div>
            <HashLink
              to="/contact?s=footer"
            >
              Contact
            </HashLink>
          </div>
          <div
            className="powered-by"
          >
            <span
              className="grey--text"
            >
              Powered By
            </span>
            <strong
              className="teal--text"
              href="#"
            >
               
              NoStack
            </strong>
          </div>
        </div>
        <div
          className="grow-6 "
        >
          <div
            className="flex footer__group-links"
          >
            <div
              className="grow-4"
            />
            <div
              className="grow-4"
            >
              <div
                className="footer__title"
              >
                About
              </div>
              <ul
                className="footer__links"
              >
                <li>
                  <HashLink
                    to="/#about-us"
                  >
                     About Us
                  </HashLink>
                </li>
                <li>
                  <HashLink
                    to="/#how-it-works"
                  >
                     How it Works
                  </HashLink>
                </li>
                <li>
                  <HashLink
                    to="/pricing"
                  >
                     Pricing
                  </HashLink>
                </li>
                <li>
                  <strong
                    aria-hidden="true"
                    className="teal--text"
                    href="#"
                    role="button"
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                    tabIndex={0}
                  >
                    Sign Up Now
                  </strong>
                </li>
              </ul>
            </div>
            <div
              className="grow-4"
            >
              <div
                className="footer__title"
              >
                Terms
              </div>
              <ul
                className="footer__links"
              >
                <li>
                   
                  <HashLink
                    to="/terms-and-conditions"
                  >
                    Terms of Use
                  </HashLink>
                </li>
                <li>
                   
                  <HashLink
                    to="/privacy-policy"
                  >
                    Privacy
                  </HashLink>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        className="text-center copyright"
      >
        Copyright Â© 2020 Pivotate
      </div>
    </div>
  </div>
</footer>
`;
